### PR TITLE
[WIP #1] Support for NSX client

### PR DESCRIPTION
**Pull request by rachappag**
Added very basic support to initialize NSX Clinet

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package govnsx
+
+import (
+	//"fmt"
+
+	"github.com/go-resty/resty"
+)
+
+type Client struct {
+	*resty.Request
```
Comment (chandidas) [inline client.go]: Add explicit variable (say, req) for readability.

File: client.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+type NsxManagerParams struct {
```
Comment (chandidas) [inline client.go]: Change name of the structure to NsxManagerConfig

File: client.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+	UserName      string
+	Password      string
+	Uri           string
+	AllowInsecssl bool
+	RestAgentName string
```
Comment (chandidas) [inline client.go]: Change member RestAgentName to UserAgentName.. for better meaning

File: client.go [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+}
+
+// NewClient creates a new client from a URL.
+func NewClient(mgrParams *NsxManagerParams) (*Client, error) {
```
Comment (chandidas) [inline client.go]: Why mgrParams are not stored locally?

**Review by chandidas — CHANGES_REQUESTED** [review]

File: client.go (hunk 1 above)
Comment (rachappag) [reply client.go]: added explicit symbolic ref

File: client.go (hunk 3 above)
Comment (rachappag) [reply client.go]: Changed

File: client.go (hunk 4 above)
Comment (rachappag) [reply client.go]: ToDo added as comment in the code

**Review by chandidas — APPROVED** [review]

**Pull request merged**